### PR TITLE
add getSignatureRegex/setSignatureRegex

### PR DESCRIPTION
**Pull request by glensc**
this is mostly to provide gateway for issues like #42

also, could you tag a release after this gets merged, thanks!

### CODE REVIEW

**Comment by glensc** [comment]
@willdurand ping?

**Comment by glensc** [comment]
@willdurand any chance to get this merged and released? so i could stop using own fork.

File: tests/EmailReplyParser/Tests/Parser/EmailParserTest.php [hunk 1]
```diff
@@ -264,6 +264,12 @@ public function testCustomQuoteHeader2()
         $this->assertTrue($fragments[1]->isQuoted());
     }
 
+    public function testCustomSigRegex()
+    {
+        $regex = $this->parser->getSignatureRegex();
+        $this->parser->setSignatureRegex($regex);
```
Comment (willdurand) [inline tests/EmailReplyParser/Tests/Parser/EmailParserTest.php]: Does this test case actually test anything?

Comment (glensc) [inline tests/EmailReplyParser/Tests/Parser/EmailParserTest.php]: seems not other than the method call doesn't fail. what you suggest to test there?

**Comment by glensc** [comment]
rebased against current master!

**Comment by willdurand** [comment]
@glensc how would you feel about giving me a hand on this project (by becoming a contributor)?

**Comment by glensc** [comment]
@willdurand sure, i can get this merged then :D

**Comment by willdurand** [comment]
perfect, but don't forget to add a test case first 😉

**Comment by glensc** [comment]
i was hoping to bypass that :smiling_imp: 

but what to test there really?
https://github.com/willdurand/EmailReplyParser/pull/45#discussion_r72622862

**Comment by willdurand** [comment]
I think you could add a test case similar to the other ones (with a fixture file) but involving a different signature since you obviously need that: it will document the use case you have that led to this patch.

**Comment by glensc** [comment]
alright, i'll take testcase from #42

**Comment by glensc** [comment]
ok, looks good?

i think 2.7.0 is on the way.

and i think i'll create `CHANGELOG.md` before tagging it. it's nice to have changelog in repo as well.

example: https://github.com/eventum/eventum/blob/master/CHANGELOG.md

File: tests/EmailReplyParser/Tests/Parser/EmailParserTest.php [hunk 2]
```diff
@@ -410,6 +410,24 @@ public function testEmailWithFairAmountOfContent()
         $this->assertRegexp('/^On Thursday/', (string) $fragments[0]);
     }
 
+    /**
+     * override regexp, not to match too greedy signature.
+     *
+     * @see https://github.com/willdurand/EmailReplyParser/pull/42
```
Comment (willdurand) [inline tests/EmailReplyParser/Tests/Parser/EmailParserTest.php]: I would simply add this:

```
// See: https://github.com/willdurand/EmailReplyParser/pull/42.
```

Comment (glensc) [reply tests/EmailReplyParser/Tests/Parser/EmailParserTest.php]: `@see` is official `@phpdoc` annotation, phpstorm didn't make links clickable in the past. seems that's not the case anymore. can drop.

http://docs.phpdoc.org/references/phpdoc/tags/see.html

File: src/EmailReplyParser/Parser/EmailParser.php [hunk 3]
```diff
@@ -141,6 +143,28 @@ public function setQuoteHeadersRegex(array $quoteHeadersRegex)
         return $this;
     }
 
+    /**
+     * @return string
+     * @since 2.7.0
```
Comment (willdurand) [inline src/EmailReplyParser/Parser/EmailParser.php]: Ah I did not pay attention to that, usually I dont use this and I dont think we should be using this. The `CHANGELOG` file (or GitHub release) is way more expressive.

Comment (glensc) [reply src/EmailReplyParser/Parser/EmailParser.php]: ok, so remove the annotation and commit to `master`?

**Pull request merged**